src/global/global_init.cc

```diff
@@ -362,9 +381,14 @@ global_init(const std::map<std::string,std::string> *defaults,
   if (g_conf()->run_dir.length() &&
       code_env == CODE_ENVIRONMENT_DAEMON &&
       !(flags & CINIT_FLAG_NO_DAEMON_ACTIONS)) {
-    int r = ::mkdir(g_conf()->run_dir.c_str(), 0755);
-    if (r < 0 && errno != EEXIST) {
-      cerr << "warning: unable to create " << g_conf()->run_dir << ": " << cpp_strerror(errno) << std::endl;
+
+    if (!fs::exists(g_conf()->run_dir.c_str())) {
+      std::error_code ec;
+      if (!fs::create_directory(g_conf()->run_dir, ec)) {
+       cerr << "warning: unable to create " << g_conf()->run_dir
+            << ec.message() << std::endl;
+      }
+      fs::permissions(g_conf()->run_dir.c_str(), fs::perms::owner_all);
```

Comment:
`0755` was the original permission -- so `owner_all | group_read | group_exec | others_write | others_exec`?

Reply:
Right, my bad. Thanks for bringing it up.